memory.go

```diff
@@ -96,6 +96,8 @@ type MemProvider struct {
 // Init initializes memory session provider.
 func (p *MemProvider) Init(maxLifetime int64, _ string) error {
 	p.lock.Lock()
+	p.list = list.New()
+	p.data = make(map[string]*list.Element)

```

Comment:
What is the fix/benefit here?

Reply:
This standardises the init function for the memprovider as compared to the other providers. 

All other providers the init function actually makes the provider ready - only the memprovider is different.

Say you wanted to have a doubly backed provider that wraps the eventual provider - for whatever reason - this glaring inconsistency breaks that.

Reply:
Gotcha, thanks!